src/metatrain/cli/train.py

```diff
@@ -406,12 +408,27 @@ def train_model(
     logger.info(
         f"Exporting model to `{output_checked}` and extensions to `{extensions_path}`"
     )
+    # get device from the model
+    device = next(
+        itertools.chain(
+            mts_atomistic_model.parameters(),
+            mts_atomistic_model.buffers(),
+        )
+    ).device
```

Comment:
Why do have to get the device from the model? It should be the same as the device we picked above?

Reply:
Unfortunately that's not the case if training is distributed. I can add a comment for now. I think there would be a way to restructure the code so that the two are the same, but it involves moving a lot of responsibility from the model to metatrain in the distributed case, which is a decision that should be discussed because it has also a lot of drawbacks

Reply:
Okay fine. But then maybe we should call it `final_device` or something to make sure that this is device after the training is performed.

Reply:
Sure, commented and renamed to `final_device`